Source/Source/Patches/InteractionWorker_RecruitAttempt_DoRecruit_Patch.cs

```diff
@@ -20,7 +20,7 @@ public static void Postfix(Pawn recruiter, Pawn recruitee, bool useAudiovisualEf
                     recruitee.SetFaction(recruiter.HostFaction);
                     if (faction != null)
                     {
-                        Find.FactionManager.Notify_PawnRecruited(faction);
+                        recruiter.HostFaction.Notify_PawnJoined(recruitee);
```

Comment:
Have you checked that this is an equivalent replacement?

Reply:
I just checked, and it appears to not be 😓. Though this method seems to be removed from FactionManager and is no longer called from DoRecruit, so I'm thinking Hospitality could do the same. See attached diff of InteractionWorker_RecruitAttempt DoRecruit, left = 1.3, right = 1.4
![DoRecruitDiff](https://user-images.githubusercontent.com/29614757/198023976-e02678a4-2c1c-45d8-83fd-0af7edffdb95.png)

Reply:
Ugh. I do not like how bloated InteractionWorker_RecruitAttempt is. It has changed a lot, I see.
The notification bit is much further down, so what you see on the right of that diff is _not_ the equivalent of the left.

Reply:
Lol, I feel you on the bloat. Is this the notification bit you're referring to? It appears in both sides with minimal change (expanded diff)
![DoRecruitDiff2](https://user-images.githubusercontent.com/29614757/198029880-85622319-950b-4863-bcfd-a62e3f25391c.png)

I also looked at Notify_PawnRecruited, it seems it's pretty short and doesn't touch the "letter" notifications the player receives.

![Notify_PawnRecruited](https://user-images.githubusercontent.com/29614757/198030327-24273e12-ca56-430f-ae31-afcf73cd4ba3.png)

I do wonder, what uses this DoRecruit patch currently? I see the wardening for guests is currently disabled with the comment "Too many issues with this" in JobGiver_Work_Patch.cs and I can't think of other cases where it'd apply (again, not educated on RimWorld modding, so please forgive my ignorance if it's showing 😅 ).

Reply:
Actually, looking at it, what I gather is that FactionManager.Notify_PawnRecruited can be removed (and you did). I'm not sure if HostFaction.Notify_PawnJoined needs to be added or is already called somewhere in the chain of functions (likely SetFaction).
So you'd have to check that it doesn't get called twice.

Reply:
You're correct, SetFaction does seem to already call this. 

![setfaction](https://user-images.githubusercontent.com/29614757/198109220-dff4d69d-1268-452f-a22f-f971dc1ccc76.png)

I've removed the call in [7cf9729](https://github.com/OrionFive/Hospitality/pull/703/commits/7cf972955daffe9601c9479cdda9f5aebf84b154).